purchase_order_product_recommendation/tests/test_recommendation.py

```diff
@@ -0,0 +1,306 @@
+# Copyright 2019 David Vidal <david.vidal@tecnativa.com>
+# Copyright 2020 Manuel Calero - Tecnativa
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from datetime import date, datetime
+
+from odoo.tests.common import SavepointCase
+
+
+class RecommendationCase(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super(RecommendationCase, cls).setUpClass()
+        cls.partner = cls.env["res.partner"].create({"name": "Mr. Odoo"})
+        cls.category_obj = cls.env["product.category"]
+        cls.categ1 = cls.category_obj.create({"name": "Test Cat 1"})
+        cls.categ2 = cls.category_obj.create({"name": "Test Cat 2"})
+        cls.product_obj = cls.env["product.product"]
+        cls.prod_1 = cls.product_obj.create(
+            {
+                "default_code": "product-1",
+                "name": "Test Product 1",
+                "categ_id": cls.categ1.id,
+                "type": "product",
+                "seller_ids": [(0, 0, {"name": cls.partner.id, "price": 5})],
+            }
+        )
+        cls.prod_2 = cls.product_obj.create(
+            {
+                "default_code": "product-2",
+                "name": "Test Product 2",
+                "categ_id": cls.categ2.id,
+                "type": "product",
+                "seller_ids": [(0, 0, {"name": cls.partner.id, "price": 10})],
+            }
+        )
+        cls.prod_3 = cls.product_obj.create(
+            {
+                "default_code": "product-3",
+                "name": "Test Product 3",
+                "categ_id": cls.categ2.id,
+                "type": "product",
+                "seller_ids": [(0, 0, {"name": cls.partner.id, "price": 7})],
+            }
+        )
+        # Warehouses
+        cls.wh1 = cls.env["stock.warehouse"].create(
+            {"name": "TEST WH1", "code": "TST1"}
+        )
+        cls.wh2 = cls.env["stock.warehouse"].create(
+            {"name": "TEST WH2", "code": "TST2"}
+        )
+        # Locations
+        location_obj = cls.env["stock.location"]
+        cls.supplier_loc = location_obj.create(
+            {"name": "Test supplier location", "usage": "supplier"}
+        )
+        cls.customer_loc = location_obj.create(
+            {"name": "Test customer location", "usage": "customer"}
+        )
+        # Create deliveries and receipts orders to have a history
+        cls.picking_obj = cls.env["stock.picking"]
+        cls.picking_1 = cls.picking_obj.create(
+            {
+                "location_id": cls.wh1.lot_stock_id.id,
+                "location_dest_id": cls.customer_loc.id,
+                "partner_id": cls.partner.id,
+                "picking_type_id": cls.wh1.out_type_id.id,
+            }
+        )
```

Comment:
Why do you need a out picking location for this module?

Reply:
They were in version 12.0 I have simply respected it

Reply:
Ok.